### PR TITLE
Boundary condition change conditional for setting temslope

### PR DESCRIPTION
**Pull request by dvalters**
Changes from `x <= 2` to `x < 1`. (and for y)

Corresponds to jmax/imax at other boundary conditions of model domain. 

Let's see if it breaks the tests....

### CODE REVIEW

**Comment by dvalters** [comment]
Passes locally:

```
basic_kga_tests.py::TestingHAILCAESAR::test_water_depths[Final water depth raster] PASSED                                                      [ 50%]
basic_kga_tests.py::TestingHAILCAESAR::test_hydrograph_lisflood[LISFLOOD hydrograph] PASSED                                                    [100%]

============================================================== 2 passed in 0.20 seconds ==============================================================
```

But not on Travis ???

File: src/catchmentmodel/LSDCatchmentModel.cpp [hunk 1]
```diff
@@ -2345,7 +2345,7 @@ void LSDCatchmentModel::flow_route()
                 (elev[x][y] + water_depth[x][y])) / DX;
 
             if (x == imax) tempslope = edgeslope;
-            if (x <= 2) tempslope = 0 - edgeslope;
```
Comment (dvalters) [inline src/catchmentmodel/LSDCatchmentModel.cpp]: Why is tempslope negative at the Western domain boundary, and positive at the Eastern domain boundary. (And similarly for N-S)?

This seems odd...

**Comment by dvalters** [comment]
@aproeme I flipped the DEM in this one so that the catchment is on the Eastern edge. Tempslope is positive and equal to edgeslope now on all the boundaries:

equal_edgeslope_flippeddems

There are slightly higher peak discharges when edgeslope is positive on the E edge, but I'm wondering if this is  actually more 'correct' since having a negative edgeslope on the boundary edge would act like a little 'speedbump' slowing the water down slightly...

**Comment by dvalters** [comment]
For reference: these are the DEMs (outlet on the West - original, outlet on the East - flipped)
Units are 'grid cells' at 50m resolution

normal_dem_boscastle
flipped_dem_boscastle

**Comment by dvalters** [comment]
Will hold off on merging this into master this until we reconcile comments from Tom:

> ...is there a reason for it being x<=2 rather than x==0? (It seems like should apply only to the leftmost column of cells, mirroring the opposite side of the domain (x == xmax)

 >> Its because the flow routing is calculated via looking at the +ve or -ve water surface slope between (x,y) and (x-1,y) and between (x,y) and (x,y-1). The DEM runs from 1 (not 0) to Xmax so to work out what goes onto row 1 (the edge of the DEM where water is effectively removed) you need to work on row2. For Xmax you look at xmax as it looks at fluxes to the left (x-1) and above (y-1). 

> Secondly, is there a reason for tempslope being negative where x<=2 (and y<=2) yet positive on the opposing edges?

>> See above. As it looks at the fluxes both ways between (eg) (x,y) and (x-1,y) then on the LH edge the slope is -ve and on the RH side its +ve. 

Maybe there is a way to get around this for the libgeodecomp port... will think about it!

**Comment by dvalters** [comment]
Rough sketch of how this looks (at least in my head... :thinking:)

img_20180704_174158924_hdr
